udf/crc32_uuid.sql

```diff
@@ -0,0 +1,108 @@
+    2647816111, 81470997, 1943803523, 3814918930, 2489596804, 225274430,
+    2053790376, 3826175755, 2466906013, 167816743, 2097651377, 4027552580,
+    2265490386, 503444072, 1762050814, 4150417245, 2154129355, 426522225,
+    1852507879, 4275313526, 2312317920, 282753626, 1742555852, 4189708143,
+    2394877945, 397917763, 1622183637, 3604390888, 2714866558, 953729732,
+    1340076626, 3518719985, 2797360999, 1068828381, 1219638859, 3624741850,
+    2936675148, 906185462, 1090812512, 3747672003, 2825379669, 829329135,
+    1181335161, 3412177804, 3160834842, 628085408, 1382605366, 3423369109,
+    3138078467, 570562233, 1426400815, 3317316542, 2998733608, 733239954,
+    1555261956, 3268935591, 3050360625, 752459403, 1541320221, 2607071920,
+    3965973030, 1969922972, 40735498, 2617837225, 3943577151, 1913087877,
+    83908371, 2512341634, 3803740692, 2075208622, 213261112, 2463272603,
+    3855990285, 2094854071, 198958881, 2262029012, 4057260610, 1759359992,
+    534414190, 2176718541, 4139329115, 1873836001, 414664567, 2282248934,
+    4279200368, 1711684554, 285281116, 2405801727, 4167216745, 1634467795,
+    376229701, 2685067896, 3608007406, 1308918612, 956543938, 2808555105,
+    3495958263, 1231636301, 1047427035, 2932959818, 3654703836, 1088359270,
+    936918000, 2847714899, 3736837829, 1202900863, 817233897, 3183342108,
+    3401237130, 1404277552, 615818150, 3134207493, 3453421203, 1423857449,
+    601450431, 3009837614, 3294710456, 1567103746, 711928724, 3020668471,
+    3272380065, 1510334235, 755167117
+  ]
+);
+CREATE TEMP FUNCTION udf_crc32_partial_1(crc INT64, value BYTES) AS (
+  (crc >> 8) ^ udf_crc32_table()[OFFSET((crc ^ TO_CODE_POINTS(value)[OFFSET(0)]) & 0xFF)]
+);
+CREATE TEMP FUNCTION udf_crc32_partial_2(crc INT64, value BYTES) AS (
+  udf_crc32_partial_1(
+    udf_crc32_partial_1(crc, SUBSTR(value, 1, 1)),
+    SUBSTR(value, 2, 1)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_partial_4(crc INT64, value BYTES) AS (
+  udf_crc32_partial_2(
+    udf_crc32_partial_2(crc, SUBSTR(value, 1, 2)),
+    SUBSTR(value, 3, 2)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_partial_8(crc INT64, value BYTES) AS (
+  udf_crc32_partial_4(
+    udf_crc32_partial_4(crc, SUBSTR(value, 1, 4)),
+    SUBSTR(value, 5, 4)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_partial_16(crc INT64, value BYTES) AS (
+  udf_crc32_partial_8(
+    udf_crc32_partial_8(crc, SUBSTR(value, 1, 8)),
+    SUBSTR(value, 9, 8)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_partial_32(crc INT64, value BYTES) AS (
+  udf_crc32_partial_16(
+    udf_crc32_partial_16(crc, SUBSTR(value, 1, 16)),
+    SUBSTR(value, 17, 16)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_partial_36(crc INT64, value BYTES) AS (
+  udf_crc32_partial_4(
+    udf_crc32_partial_32(crc, SUBSTR(value, 1, 32)),
+    SUBSTR(value, 33, 4)
+  )
+);
+CREATE TEMP FUNCTION udf_crc32_uuid(value BYTES) AS (
+  udf_crc32_partial_36(0xFFFFFFFF, value) ^ 0xFFFFFFFF
```

Comment:
I chose to only fully implement 36-byte crc32 because every instance of client id in mozilla-pipeline-schemas is validated as that length, except in the metadata schemas.